papers/scipy/scipy.rst

```diff
@@ -0,0 +1,1706 @@
+the initial conditions of the forward and backward passes.
+
+Bandpass filter
+---------------
+
+In this example, we will use synthetic data to demonstrate a
+bandpass filter.  We have 0.03 seconds of data sampled at
+4800 Hz.  We want to apply a bandpass filter to remove frequencies
+below 400 Hz or above 1200 Hz.
+
+Just like we did for the lowpass filter, we define two functions that
+allow us to create and apply a Butterworth bandpass filter with the
+frequencies given in Hz (or any other units).  The functions take care
+of scaling the values to the normalized range expected by the SciPy function
+``scipy.signal.butter``.
+
+.. code-block:: python
+
+    from scipy.signal import butter, sosfilt
+
+    def butter_bandpass(lowcut, highcut, fs, order):
+        nyq = 0.5 * fs
+        low = lowcut / nyq
+        high = highcut / nyq
+        sos = butter(order, [low, high], btype='band',
+                     output='sos')
+        return sos
+
+    def butter_bandpass_filt(data, lowcut, highcut,
+                             fs, order):
+        sos = butter_bandpass(lowcut, highcut, fs,
+                              order)
+        y = sosfilt(sos, data)
+        return y
+
+First, we'll take a look at the frequency response of the Butterworth
+bandpass filter with order 3, 6, and 12.  The code that generates
+Figure :ref:`fig-bandpass-example-response` demonstrates the use of
+``scipy.signal.sosfreqz``:
+
+.. code-block:: python
+
+    for order in [3, 6, 12]:
+        sos = butter_bandpass(lowcut, highcut, fs, order)
+        w, h = sosfreqz(sos, worN=2000)
+        plt.plot((fs*0.5/np.pi)*w, abs(h), 'k',
+                 alpha=(order+1)/13,
+                 label="order = %d" % order)
+
+.. figure:: figs/bandpass_example_response.pdf
+
+    Amplitude response for a Butterworth bandpass filter
+    with several different orders.
+    :label:`fig-bandpass-example-response`
+
+Figure :ref:`fig-bandpass-example-signals` shows the input signal and
+the filtered signal.  The order 12 bandpass Butterworth filter
+was used.  The plot shows the input signal `x`; the filtered signal
+was generated with
+
+.. code-block:: python
+
+    y = butter_bandpass_filt(x, lowcut, highcut, fs,
+                             order=12)
+
+where ``fs = 4800``, ``lowcut = 400`` and ``highcut = 1200``.
+
+.. figure:: figs/bandpass_example_signals.pdf
+
+    Original noisy signal and the filtered signal.
+    The order 12 Butterworth bandpass filter shown in
+    Figure :ref:`fig-bandpass-example-response` was used.
+    :label:`fig-bandpass-example-signals`
+
+Filtering a long signal in batches
+----------------------------------
+
+We will again use synthetic data generated by the same function
+used in the previous example, but for a longer time interval.
+
+This example shows how the state of the IIR filter can be saved
+and restored, so a filter can be applied to a long signal in batches.
```

Comment:
It wasn't immediately clear to me why this was done - first thought was to speed up the calculation, but it's because you want to filter a signal while the data is still being acquired (right?). Worth pointing out.

Reply:
A edited this section to (hopefully) clarify the motivation.